storage/Devices/LvmLvImpl.cc

```diff
@@ -245,33 +250,67 @@ namespace storage
 	    switch (lv.lv_type)
 	    {
 		case LvType::NORMAL:
-		case LvType::THIN_POOL:
 		case LvType::RAID:
+		case LvType::CACHE:
+		case LvType::WRITECACHE:
 		{
-		    LvmVg* lvm_vg = LvmVg::Impl::find_by_uuid(prober.get_system(), lv.vg_uuid);
-		    lvm_lv = lvm_vg->create_lvm_lv(lv.lv_name, lv.lv_type, lv.size);
+		    // A cache is private if uses as a thin-pool.
+
+		    if (lv.role == CmdLvs::Role::PUBLIC)
+		    {
+			LvmVg* lvm_vg = LvmVg::Impl::find_by_uuid(system, lv.vg_uuid);
```

Comment:
There appears to be an awful lot of code duplication here; this sequence repeats 4 times.

This calls for factoring out into a separate function that accepts the two parameters that are different: The UUID to find and the LvType to create.

Reply:
Unfortunately it is a bit more different. For thins the parent of the logical volume is not the volume group but the thin pool, so an object of a different type. So at least two helper functions would be needed.

Reply:
OK then.